src/backend/marsha/core/api.py

```diff
@@ -76,24 +74,26 @@ def update_state(request):
     # Update the object targeted by the "object_id" and "resource_id"
     model = apps.get_model(app_label="core", model_name=key_elements["model_name"])
 
-    update_parameters = {}
-    if serializer.validated_data["state"] == defaults.READY:
-        update_parameters["uploaded_on"] = key_elements["uploaded_on"]
-        if hasattr(model, "extension"):
-            update_parameters["extension"] = key_elements.get("extension")
-
-    updated = model.objects.filter(id=key_elements["object_id"]).update(
-        **{
-            # Only update `uploaded_on` if the upload was actually successful
-            **update_parameters,
-            "upload_state": serializer.validated_data["state"],
-        }
+    extra_parameters = serializer.validated_data["extraParameters"]
+    if serializer.validated_data["state"] == defaults.READY and hasattr(
+        model, "extension"
+    ):
+        extra_parameters["extension"] = key_elements.get("extension")
```

Comment:
It took m e some time to re-understand this "extension" thing... I think it requires a comment to explain why we get an extension from the lambda!

Reply:
The extension comes from the s3 key and this key is generated by the back application in the method `DocumentViewSet.initiate_upload` but I will add a comment.